### PR TITLE
WRQ-6438: Migrate deprecated rules to @stylistic/js

### PR DESCRIPTION
**Pull request by hong6316**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [ ] I have run automated CLI testing and it is passed
* [x] Documentation was added or is not needed
* [ ] This is an API-breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
https://eslint.org/blog/2023/10/deprecating-formatting-rules/
The next minor release of ESLint(Eslint v8.53.0) will deprecate core formatting rules.

In https://eslint.org/blog/2023/10/deprecating-formatting-rules/#what-you-should-do-instead ,
Eslint suggests using a source code formatter (e.g. prettier, dprint) or `@stylistic/eslint-plugin-js`.
`@stylistic/eslint-plugin-js` contain the deprecated formatting rules from the ESLint core respectively.
Because the above module matches exactly same with the existing API, this module was selected instead of the source code formatter.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Migrate deprecated rules to `@stylistic/eslint-plugin-js`

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
When checking the commit below, `no-extra-semi` and `no-mixed-spaces-and-tabs` rules still exist in eslint v8 and will be deprecated starting from eslint v9(https://github.com/eslint/eslint/commit/6ee3e9eb5df7bdfdaa1746214793ed511112be76).
In this PR, I will keep current this states.

### Links
[//]: # (Related issues, references)
WRQ-6438

### Comments
Enact-DCO-1.0-Signed-off-by: Taeyoung Hong (taeyoung.hong@lge.com)

### CODE REVIEW

**Comment by seunghoh** [comment]
@hong6316 Please update the PR description why you choose @stylistic/js

**Comment by hong6316** [comment]
> @hong6316 Please update the PR description why you choose @stylistic/js

OK, I wrote the info on `Issue Resolved / Feature Added`

**Review by mmyelyn — APPROVED** [review]
Good. But I think in the long run it's better to use prettier. This is because forcing formatting when saving code is more convenient than notifying the code of a Lint warning.

File: index.js [hunk 1]
```diff
@@ -113,7 +108,14 @@ module.exports = {
 		'no-var': 'warn',
 		'require-yield': 'off',
 		'use-isnan': 'warn',
-		'wrap-iife': ['error', 'inside'],
+
+		// @stylistic/js plugin https://github.com/eslint-stylistic/eslint-stylistic
+		'@stylistic/js/new-parens': 'warn',
+		'@stylistic/js/no-extra-semi': 'off',
```
Comment (juwonjeong) [inline index.js:114]: When checking the following rules, the result of the `enact lint` differ between develop branch and the feature branch.
'@stylistic/js/no-extra-semi': 'off'
'@stylistic/js/no-mixed-spaces-and-tabs': ['warn', 'smart-tabs'].
Could you please check it?

Comment (hong6316) [reply index.js:114]: When checking the commit below, these two rules still exist in eslint v8 and will be deprecated starting from eslint v9(https://github.com/eslint/eslint/commit/6ee3e9eb5df7bdfdaa1746214793ed511112be76).
In this PR, I will restore the code.

**Review by juwonjeong — APPROVED** [review]
LGTM

**Pull request merged**